### ISSUE TITLE
[CRITICAL] (LyShine) InGame console does not receive input when UI canvas visible

### ISSUE THREAD

**Issue by AMZN-alexpete** [o3de/o3de#2712]
**Describe the bug**
I'm not 100% what the cause of this bug is but it happens 100% of the time in our game jam.  After the game level is open and a UI menu is visible it becomes very difficult - sometimes impossible - to type in the in-game console window to run commands. My theory is - when you have a UI canvas in your game that consumes all input and it is active, the in-game console does not receive or process input

**To Reproduce**
1.  Create a ui canvas that consumes all input (option on canvas properties)
2.  Load the canvas automatically in the level using a ui canvas asset reference component
3. Load the level in the launcher
4. Open the console window and try to run commands

**Expected behavior**
The input from the keyboard appears in the console and you can run commands like 'quit' or 'LoadLevel'

**Screenshots**
n/a

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version da47435

**Additional context**
n/a

**Comment by Amzn-Catdo** [comment]
Same issue?
https://github.com/o3de/o3de/issues/2488
https://github.com/o3de/o3de/issues/2740
https://github.com/o3de/o3de/issues/2515

**Comment by Amzn-Catdo** [comment]
After the fix went in for input issues with UI Canvas, the other issues were closed but upon checking this issue, I could still reproduce so it is unrelated to the fix. 
What I noticed was that I could type something in the console but when I press enter it doesn't do anything. The backspace button to delete also doesn't work nor does any mouse clicking.
I went back to the UI Editor and disabled "Consume all input" then I save the ui canvas and the level. Open up game launcher and load the level. Press ` to open the console and was able to type things and delete it also click on the clear button and option button whereas when I had the "Consume all input" enabled, I was not able to delete nor click on any of those buttons on the console
![image](https://user-images.githubusercontent.com/82239748/129282186-a211a394-a55f-49a6-9191-1c398ea836d8.png)
